std/typecons.d

```diff
@@ -1732,8 +1761,36 @@ succeeds, $(D this) becomes non-null.
     }
 
 /**
+    If this `Nullable` wraps a type that already has a null value
+    (such as a pointer), then assigning the null value to this
+    `Nullable` is no different than assigning any other value of
+    type `T`, and the resulting code will look very strange. It 
+    is strongly recommended that this be avoided by instead using
+    the version of `Nullable` that takes an additional `nullValue`
+    template argument.
+ */
+unittest
+{
+    //Passes
+    Nullable!(int*) npi;
+    assert(npi.isNull);
+    
+    //Passes?!
+    npi = null;
+    assert(!npi.isNull);
+}
+
+/**
 Gets the value. $(D this) must not be in the null state.
 This function is also called for the implicit conversion to $(D T).
+
+Throws:
+    A $(CXREF exception, AssertError) in the case where `isNull` 
```

Comment:
I think `assert`s should be left as implementation details. The precondition is already mentioned in the documentation text; I think the implementation should be free to do whatever it wants if the precondition is not met (it is, after all, a bug in the caller's code).